From b56451ec2c44a4efda7dc7a18ab1b946685e4deb Mon Sep 17 00:00:00 2001
From: Robert Michelsen <robert.michelsen@gmx.net>
Date: Wed, 19 Nov 2025 14:23:58 +0100
Subject: [PATCH] tools/makedep: Ignore erroneous -D argument.

Fixes pkg-config CFLAGS containing '-Dvar'

 ../tools/makedep  -C/mainline-src-1.7.8/tools -Smainline-src-1.7.8 -T..  -I/usr/include/freetype2
      -I/usr/include/libpng16 -DWITH_GZFILEOP -I/usr/include/harfbuzz -I/usr/include/glib-2.0
      -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-6  fnt2fon.c make_ctests.c make_xftmpl.c makedep.c sfnt2fnt.c
                winemaker.de.UTF-8.man.in winemaker.fr.UTF-8.man.in winemaker.man.in
 Unknown option '-DWITH_GZFILEOP'
---
 tools/makedep.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/makedep.c b/tools/makedep.c
index 1d997623143..6e68454e13e 100644
--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -1520,6 +1520,8 @@ static void parse_option( const char *opt )
     case 'x':
         if (opt[2]) strarray_add( &object_extensions, xstrdup( opt + 2 ));
         break;
+    case 'D':
+        break;
     default:
         fprintf( stderr, "Unknown option '%s'\n%s", opt, Usage );
         exit(1);
-- 
2.51.1

